samples/msmq/sql-bridge/Core_5/SqlBridge/MsmqReceiver.cs

```diff
@@ -62,15 +70,27 @@ public bool Handle(TransportMessage message)
         // a valid Guid and if not, only then create a valid Guid. This check
         // is important as it affects the retries if the message is rolled back.
         // If the Ids are different, then the recoverability won't know its the same message.
-        Guid newGuid;
-        if (!Guid.TryParse(msmqId, out newGuid))
+        Guid msmqGuid;
+        if (!Guid.TryParse(msmqId, out msmqGuid))
         {
             headers["NServiceBus.MessageId"] = GuidBuilder.BuildDeterministicGuid(msmqId).ToString();
         }
-        log.Info("Forwarding message to all the SQL subscribers via a Publish");
-        message.TimeToBeReceived = TimeSpan.MaxValue;
-        publisher.Publish(message, new PublishOptions(eventTypes.First()));
-        return true;
+
+        var sqlMessage = new TransportMessage(headers["NServiceBus.MessageId"], headers);
```

Comment:
i think this would be better as 

```

        return new TransportMessage(headers["NServiceBus.MessageId"], headers)
        {
            Body = msmqMessage.Body,
            TimeToBeReceived = (msmqMessage.TimeToBeReceived < Message.InfiniteTimeout) ? msmqMessage.TimeToBeReceived : TimeSpan.MaxValue
        };
```